pilot/pkg/model/service.go

```diff
@@ -593,6 +612,29 @@ func (ep *IstioEndpoint) CmpOpts() []cmp.Option {
 	return istioEndpointCmpOpts
 }
 
+// SortIstioEndpointByAddresses returns the IstioEndpoint address slices sorted by its addresses.
+func (ep *IstioEndpoint) SortIstioEndpointByAddresses() []string {
+	if ep == nil {
+		return nil
+	}
+
+	addrSlice := ep.Addresses
+	sort.Sort(sort.Reverse(sort.StringSlice(addrSlice)))
```

Comment:
not clear why we reverse.

Also we are mutating in place AND returning - seems weird?

Reply:
I don't see how we can mutate without race conditions.

Reply:
^ all these concerns apply to Key as well

Reply:
Hi @howardjohn, based on above comment, I remove function of `SortIstioEndpointByAddresses` and add new function of `IsEqualIstioEndpoint`.